### ISSUE TITLE
Empty Catch block not allowed

### ISSUE THREAD

**Issue by carloscds** [code-cracker/code-cracker#15]
``` csharp
try 
{
    // do something
}
catch
{
}
```

Becomes (Fix 1:)

``` csharp
  // do something
```

Becomes (Fix 2:)

``` csharp
 // TO DO: Consider reading MSDN Documentation about how to use Try...Catch
 // http://msdn.microsoft.com/en-us/library/0yd65esw.aspx
 // do something
```

Becomes (Fix 3:)

``` csharp
try 
{
    // do something
}
catch (Exception e)
{
    throw;
}
```

**Comment by giggio** [comment]
@code-cracker/owners I am changing this from `Error` to `Warning`. I have found that it is very common to have that while I analyzed https://github.com/dotnet/corefx. I am doing this work on #100.
If you don't fell it is ok let me know.

**Comment by ljani** [comment]
How do I suppress this with a comment as in "fix 3"? The following code produces a `CC0004` for me:

```csharp
try
{
    Directory.Delete(myDirectory, true);
}
catch (DirectoryNotFoundException)
{
    // Ignored
}
```

**Comment by carloscds** [comment]
@ljani Because you don't have "code" on catch, only a comment.

**Comment by ljani** [comment]
@carloscds Yes, I understand the comments are not statements and the analyzer is only checking if there are any statements in the block. Let me rephrase my question, is there any interest implementing [the fix 3 suggested by @luisrudge](https://github.com/code-cracker/code-cracker/issues/15#issuecomment-63104708). In my opinion, it's a correct way to fix this warning in my case.

`#pragma warning disable` is not a nice solution in this case. I guess I'm back to the `UNREFERENCED_PARAMETER()`-ish days :)

Also in my opinion, the correct way to delete a directory is to `try` it instead of checking if it exists and then deleting it, because of race conditions.

**Comment by ljani** [comment]
@carloscds Well, how would you implement the method, if I need to 1) run that code before the logging functionality is bootstrapped (eg. delete the logs folder) and 2) swallow `DirectoryNotFoundException` and let others pass?

**Comment by carloscds** [comment]
@ljani Maybe we can implement like an option (fix 3), but user decide. What do you think about ?

**Comment by ljani** [comment]
Sure, that works for me. It's already pretty widespread standard to use comments to document why the catch block is empty. For example ReSharper does this.

**Comment by giggio** [comment]
@carloscds this should be a new issue. This one is done. A new issue with a clear proposal of what needs to be done should be created.

**Comment by carloscds** [comment]
@giggio Maybe, or a new option at same code.
